src/query/api/v1/handler/prometheus/remote/write.go

```diff
@@ -99,14 +101,33 @@ func (h *PromWriteHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err := h.write(r.Context(), req)
-	if err != nil {
+	batchErr := h.write(r.Context(), req)
+	if batchErr != nil {
+		numBadRequest := 0
+		errs := batchErr.Errors()
+		lastErr := batchErr.LastError()
+		for _, err := range errs {
+			switch {
+			case client.IsBadRequestError(err):
+				fallthrough
```

Comment:
Might be worth having a different output error here if it's bad requests vs invalid params specifically?

Reply:
We need both to be HTTP 400 (since local aggregation can also cause a "too old" error, which will be invalidparams not a client bad request error).

The error actually returned to the client I haven't changed, it will always be the "last error".